src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -318,7 +318,9 @@ protected override void Free()
             ArrayBuilder<(BoundReturnStatement, TypeWithAnnotations)> returnTypesOpt,
             VariableState initialState,
             Dictionary<BoundExpression, (NullabilityInfo, TypeSymbol)> analyzedNullabilityMapOpt)
-            : base(compilation, symbol, node, EmptyStructTypeCache.CreatePrecise(), trackUnassignments: true)
+            // Members of variables are tracked up to a fixed depth, to avoid cycles. The
+            // maxSlotDepth value is arbitrary but large enough to allow most scenarios.
+            : base(compilation, symbol, node, EmptyStructTypeCache.CreatePrecise(), trackUnassignments: true, maxSlotDepth: 5)
```

Comment:
>maxSlotDepth: 5 [](start = 110, length = 15)

It looks like this change has far reaching impact that goes way beyond what the PR title says: "Copy nullable state of members unconditionally up to a certain depth." In particular, it looks like it limits tracking of everything to 5 levels of nesting in all scenarios. Even those that are not involved in copying states. Even for structure fields that never had problems with circularity and are handled by definite assignment check to any depth. I do not think this is the right thing to do.

Reply:
Logged [https://github.com/dotnet/roslyn/pull/35720](https://github.com/dotnet/roslyn/pull/35720)

---
In reply to: [284911184](https://github.com/dotnet/roslyn/pull/35720#discussion_r284911184) [](ancestors = 284911184)

Reply:
Looks like copy/paste error. You probably meant https://github.com/dotnet/roslyn/issues/35773